packages/channel-client/src/utils.ts

```diff
@@ -1,3 +1,25 @@
-export function sleep(time: number): Promise<void> {
-  return new Promise(resolve => setTimeout(resolve, time));
+import {hashMessage} from 'ethers/utils';
+
+import {AllocationItem, Allocation, Participant} from './types';
+
+export function calculateChannelId(
+  participants: Participant[],
+  allocations: Allocation[],
+  appDefinition: string,
+  appData: string
+): string {
+  // This `message` construction revolves around `hashMessage` taking in a
+  // string but since the concat of the arguments are unique to each channel
+  // the channel ID should be different as a result.
```

Comment:
Why should it be different per `allocations` or `appData`? Channel ID is unique on `participants`, `channelNonce`, and `appDefinition` only, and there is a specific calculation for it. I think there are helper utils in `devtools`.